frontend/public/module/k8s/get-resources.ts

```diff
@@ -102,6 +105,35 @@ export const pluralizeKind = (kind: string): string => {
   return pluralized;
 };
 
+export const getModelExtensionMetadata = (
+  extensions: LoadedExtension<ModelMetadata>[],
+  group: string,
+  version?: string,
+  kind?: string,
+) => {
+  const groupVersionKindMetadata = extensions
+    .filter(
+      ({ properties }) =>
+        properties?.model?.group === group &&
+        properties?.model?.kind === kind &&
+        properties?.model?.version === version,
+    )
+    .map((e) => e?.properties);
+  const groupKindMetadata = extensions
+    .filter(
+      ({ properties }) => properties?.model?.group === group && properties?.model?.kind === kind,
+    )
+    .map((e) => e?.properties);
+  const groupMetadata = extensions
+    .filter(({ properties }) => properties?.model?.group === group)
```

Comment:
Need to ensure that the extension model only has a group:
```suggestion
    .filter(({ properties }) => properties.model.kind == null && properties.model.version == null && properties.model.group === group)
```